spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/filter/factory/rewrite/ModifyRequestBodyGatewayFilterFactory.java

```diff
@@ -71,8 +73,9 @@ public GatewayFilter apply(Config config) {
 
 				// TODO: flux or mono
 				Mono<?> modifiedBody = serverRequest.bodyToMono(inClass)
-						// .log("modify_request_mono", Level.INFO)
-						.flatMap(o -> config.rewriteFunction.apply(exchange, o));
+						.map((Function<Object, Optional<?>>) Optional::of).defaultIfEmpty(Optional.empty())
+						.flatMap((Function<Optional<?>, Mono<?>>) o ->
+								(Mono<?>) config.rewriteFunction.apply(exchange, o.orElse(null)));
```

Comment:
as for me, it is a good idea to give an ability to specify body in case if it is absent per route
with such implementation, everyone will need to create own configuration and inject it in the rewrite function.

Reply:
I don`t understand `give an ability to specify body in case if it is absent per route
with such implementation`.
Could you explain in more detail?

Reply:
@Fetsivalen
I dont understand 'give an ability to specify body in case if it is absent per route with such implementation.-
Could you explain in more detail?

Reply:
The main idea is to give an ability to specify some default body for the empty body case.
for example "{}" for all JSON based requests.
to do the simple rewrite function and don't write code like:
```
if (body == null) {
// do this
} else {
//do that
}
```

It seems to me that it will be much simpler to have a configuration
```
        .route("test_modify_request_body",
            r -> r.order(-1).host("**.modifyrequestbody.org")
                   .filters(f -> f.modifyRequestBody(
                           String.class,
                           String.class,
                           MediaType.APPLICATION_JSON_VALUE,
                           // for non empty body
                           (serverWebExchange, aVoid) -> {return Mono.just("modifyrequest");}),
                           // for empty body
                           (serverWebExchange) -> {return Mono.just("modifyrequest");}) 
                        )
                        .uri(uri))
```
and just leave only rewrite logic inside the rewrite function.
you're still be  able to use the same function, but someone who'll need to add fallback value will have this ability.

As for me, it'll be sufficient to have just empty body supplier, but for modifying response body I were asked to add the separate rewrite function, so I think it makes sense to do the same here